can/zynqmp-can/zynqmp-can-test.c

```diff
@@ -0,0 +1,401 @@
+	if (ret == -EAGAIN) {
+		printf("zynqmp-can-test: As expected, failed to send all frames, sent only %u frames\n", sendFrames);
+	}
+	else {
+		printf("zynqmp-can-test: Unexpected test result, error code %i, sent %u frames\n", ret, sendFrames);
+	}
+
+	return ret;
+}
+
+/**
+ * Receives CAN frames from the RX FIFO once.
+ */
+static int zynqmpCanTest_RecvOneTime(oid_t *oid)
+{
+	/* Read all frames from RX FIFO */
+	zynqmp_canFrame buf[64] = { 0 };
+	uint32_t recvFrames = 0;
+	time_t time = 0; /* Wait indefinitely */
+	int ret = zynqmp_canRecv(oid, buf, (sizeof(buf) / sizeof(buf[0])), time, &recvFrames);
+	if (ret == 0) {
+		printf("zynqmp-can-test: Successfully read %u CAN frames\n", recvFrames);
+		for (uint32_t i = 0; ((i < (sizeof(buf) / sizeof(buf[0]))) && (i < recvFrames)); i++) {
+			printf("CAN frame ID: 0x%x DLC: %u Data: ", buf[i].id, buf[i].len);
+			for (int n = 0; n < buf[i].len; n++) {
+				printf("0x%02X ", buf[i].payload.bytes[n]);
+			}
+			printf("\n");
+		}
+	}
+	else {
+		printf("zynqmp-can-test: Failed to receive frames, error %i\n", ret);
+	}
+
+	return ret;
+}
+
+/**
+ * Receives CAN frames from the RX FIFO with a timeout.
+ */
+static int zynqmpCanTest_RecvOneTimeTimeout(oid_t *oid)
+{
+	/* Read all frames from RX FIFO */
+	zynqmp_canFrame buf[64] = { 0 };
+	uint32_t recvFrames = 0;
+	time_t time = 5 * 1000 * 1000; /* Wait a few seconds */
+	int ret = zynqmp_canRecv(oid, buf, (sizeof(buf) / sizeof(buf[0])), time, &recvFrames);
+	if (ret == 0) {
+		printf("zynqmp-can-test: Successfully read %u CAN frames\n", recvFrames);
+		for (uint32_t i = 0; ((i < (sizeof(buf) / sizeof(buf[0]))) && (i < recvFrames)); i++) {
+			printf("CAN frame ID: 0x%x DLC: %u Data: ", buf[i].id, buf[i].len);
+			for (int n = 0; n < buf[i].len; n++) {
+				printf("0x%02X ", buf[i].payload.bytes[n]);
+			}
+			printf("\n");
+		}
+	}
+	else if (ret == -ETIME) {
+		printf("zynqmp-can-test: Received zero frames, timeout\n");
+	}
+	else {
+		printf("zynqmp-can-test: Failed to receive frames, error %i\n", ret);
+	}
+
+	return ret;
+}
+
+/**
+ * Receives CAN frames in a loop with a timeout.
+ * Displays received frame statistics in real time.
+ */
+static int zynqmpCanTest_RecvLoop(oid_t *oid)
+{
+	int ret = 0;
+
+	/* Save previous termios state */
+	struct termios termiosCfg;
+	if (tcgetattr(STDIN_FILENO, &termiosCfg) < 0) {
+		fprintf(stderr, "cantest: tcgetattr() failed\n");
+		return -EXIT_FAILURE;
+	}
+
+	/* Modify termios state */
+	struct termios termiosCfgNew;
+	memcpy(&termiosCfgNew, &termiosCfg, sizeof(struct termios));
+	termiosCfgNew.c_lflag &= ~ICANON;
+	termiosCfgNew.c_lflag &= ~ECHO;
+	termiosCfgNew.c_cc[VMIN] = 1;
+	if (tcsetattr(STDIN_FILENO, TCSANOW, &termiosCfgNew) < 0) {
+		fprintf(stderr, "cantest: failed to set termios mode\n");
+		return -EXIT_FAILURE;
+	}
+
+	if (mutexCreate(&common.mutex_stats_access) != 0) {
+		fprintf(stderr, "cantest: Failed to create mutex\n");
+		ret = -EXIT_FAILURE;
+		goto restore_termios;
```

Comment:
`goto` is forbidded both by MISRA and our coding convention: https://docs.phoenix-rtos.com/latest/coding/index.html